File: E_Commerce/E_Commerce/E_Commerce.Dal/Entities/Customer.cs
namespace E_Commerce.Dal.Entities;

public class Customer
{
    public long CustomerId { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Email { get; set; }
    public string PhoneNumber { get; set; }
    public Cart Cart { get; set; }
    public List<Order> Orders { get; set; }
    public List<Card> Cards { get; set; }
}

File: Client/src/app/services/models/Customer.ts
import { Card } from "../../api/interfaces/Card.Interface";
import { Cart } from "../../api/interfaces/Cart.Interface";
import { CardModel } from "./Card";
import { CartModel } from "./Cart";
import { OrderModel } from "./Order";

export enum UserRole {
  Seller = 'Seller',
  Buyer = 'Buyer',
  Admin = 'Admin',
}




export class CustomerModel {
  
  customerId: number = 0; 
  firstName: string = ''; 
  lastName: string = '';
  email: string = '';
  phoneNumber: string = '';
  role: UserRole = UserRole.Admin; // Initialize with a default enum value
  cart: Cart = cart; // This might still error if Cart isn't optional or initialized
  orders: OrderModel[] = []; // Initialize with an empty array
  cards: Card[] = [];   // Initialize with an empty arra
}